File: 2014/nazionali/spiedini/att/grader.pas
uses spiedini;

const MAXN = 20000000;
var N, K, i   : longint;
    s         : array[0..MAXN-1] of longint;
    fin, fout : text;

begin
    assign(fin, 'input.txt'); reset(fin);
    assign(fout, 'output.txt'); rewrite(fout);
    read(fin, N, K);
    for i:=0 to N-1 do read(fin, s[i]);
    writeln(fout, solve(N, K, s));
    close(fin);
    close(fout);
end.

File: 2014/nazionali/spiedini/att/grader.c
#include <stdio.h>
#include <assert.h>
#define MAXN 20000000

static int N, K, i;
static int s[MAXN];

int solve(int N, int K, int* S);

static FILE *fin, *fout;

int main(){
    fin = fopen("input.txt", "r");
    fout = fopen("output.txt", "w");
    assert(2 == fscanf(fin, "%d%d", &N, &K));
    for(i=0; i<N; i++) assert(1 == fscanf(fin, "%d", &s[i]));
    fprintf(fout, "%d\n", solve(N, K, s));
    fclose(fin);
    fclose(fout);
    return 0;
}
